src/test/java/edu/ucsb/cs156/happiercows/controllers/CommonsControllerTests.java

```diff
@@ -53,13 +54,29 @@
   @WithMockUser(roles = { "ADMIN" })
   @Test
   public void createCommonsTest() throws Exception {
-    Commons expectedCommons = Commons.builder().name("TestCommons").build();
+    String testName = "TestCommons";
+    double testCowPrice = 10.4;
+    double testMilkPrice = 5.6;
+    double testStartingBalance = 50.0;
+    Date testStartDate = new Date(1646687907L);
+    Date testEndDate = new Date(1846687907L);
+    Commons expectedCommons = Commons.builder()
+            .name(testName)
+            .cowPrice(testCowPrice)
+            .milkPrice(testMilkPrice)
+            .startingBalance(testStartingBalance)
+            .startDate(testStartDate)
+            .endDate(testEndDate)
+            .build();
     ObjectMapper mapper = new ObjectMapper();
     String requestBody = mapper.writeValueAsString(expectedCommons);
     when(commonsRepository.save(any())).thenReturn(expectedCommons);
 
     MvcResult response = mockMvc
-        .perform(post("/api/commons/new?name=TestCommons").with(csrf()).contentType(MediaType.APPLICATION_JSON)
+        .perform(post(
+                String.format("/api/commons/new?name=%s?cowPrice=%f?milkPrice=%f?startingBalance=%f?startDate=%s?endDate=%s",
```

Comment:
Building URLs with String.format is generally frowned upon.  A better choices is to use the techniques for inserting variables into URIs that we saw in team01, team02, team03.

For instance:


https://github.com/ucsb-cs156-w22/STARTER-team03/blob/a357367a804aa3fc4290a01bd2309a91a2b41c0d/src/main/java/edu/ucsb/cs156/example/services/RedditPostService.java#L49      

```java
    public static final String ENDPOINT = "https://www.reddit.com/r/{subreddit}.json";
   
     Map<String, String> uriVariables = Map.of("subreddit", subreddit);
     
      ResponseEntity<String> re = restTemplate.exchange(ENDPOINT, HttpMethod.GET, entity, String.class, uriVariables);
       
```

Reply:
The reason I didn't initially do it this way was because the `CommonsControllerTests` class doesn't have a `restTemplate` instance variable, and I also wasn't sure how to incorporate mockito into the `restTemplate.exchange()` function call. Is there a way to basically call `UriTemplateHandler.expand()` in a static context so I can just get the formatted URI as a string and send it to mockito?

Reply:
Got it... as we discussed live in class, if this is in a test, it's probably ok.  If it were in a service, we'd want to use the uriVariables mechanism.